### DISCUSSION TITLE
named branch alias

### DISCUSSION THREAD

**Discussion by staabm** [composer/composer#10787] [Q&A / Support]
in one of our projects we are using release branches, and don't create tags.

therefore the consuming packages define a constraint like
`our/framework: dev-2022R01`

when developing the next version of `our/framework` in the `master`-branch, it would be great we could use a branch alias, so early adopters can try the next framework version easily.

I tried several things which did not work:

---- 

take 1

```
  "extra": {
    "branch-alias": {
      "dev-master": "CLX6_2022R02-dev"
    }
  },
  ```

=> 

```
$ composer validate
./composer.json is valid, but with a few warnings
# General warnings
- extra.branch-alias.dev-master : the target branch (CLX6_2022R02-dev) must be a parseable number like 2.0-dev

```
---- 

take 2

```
  "extra": {
    "branch-alias": {
      "dev-master": "dev-CLX6_2022R02"
    }
  },
  ```

=>

```
$ composer validate
./composer.json is valid, but with a few warnings
# General warnings
- extra.branch-alias.dev-master : the target branch (dev-CLX6_2022R02) must end in -dev
```

----

is this supported somehow?

**Comment by Seldaek** [comment]
You could also use numeric branch names for releases and everything becomes simpler (IMO..), i.e.:

- `dev-master` => `2022.2-dev` alias
- Use `2022.1` instead of `2022R01` for current release's branch name
- Then people can require `2022.2.*@dev` for now for the upcoming, and it will keep working when a 2022.2 branch is created.
